syntax = "proto3";

package product;

service ProductService {
  rpc GetProduct(GetProductRequest) returns (ProductResponse);
  rpc GetProducts(GetProductsRequest) returns (ProductsResponse);
  rpc CreateProduct(CreateProductRequest) returns (ProductResponse);
  rpc UpdateProduct(UpdateProductRequest) returns (ProductResponse);
  rpc DeleteProduct(DeleteProductRequest) returns (DeleteProductResponse);
}

message GetProductRequest {
  int32 product_id = 1;
}

message GetProductsRequest {
  // Empty request for getting all products
}

message CreateProductRequest {
  string name = 1;
  double price = 2;
  string description = 3;
}

message UpdateProductRequest {
  int32 product_id = 1;
  string name = 2;
  double price = 3;
  string description = 4;
}

message DeleteProductRequest {
  int32 product_id = 1;
}

message ProductResponse {
  int32 id = 1;
  string name = 2;
  double price = 3;
  string description = 4;
  string created_at = 5;
}

message ProductsResponse {
  repeated ProductResponse products = 1;
}

message DeleteProductResponse {
  bool success = 1;
  string message = 2;
}

